odoo/addons/base/models/res_currency.py

```diff
@@ -247,16 +248,24 @@ class CurrencyRate(models.Model):
         ('currency_rate_check', 'CHECK (rate>0)', 'The currency rate must be strictly positive.'),
     ]
 
+    def _get_default_rate(self):
+        latest_rate = self._get_latest_rate(self.env.context.get('active_id'))
+        return latest_rate and latest_rate.rate or 1.0
```

Comment:
why not 
```
@api.depends('currency_id', 'company_id', 'name')
def _compute_rate(self):
    for curr in self:
        # Note: this way is not efficient because it makes a search inside a loop.
        curr.rate = curr._get_latest_rate().rate
```

Reply:
with `store=True` and `readonly=False` ofc